paimon-core/src/main/java/org/apache/paimon/table/source/snapshot/IncrementalStartingScanner.java

```diff
@@ -46,6 +46,9 @@ public IncrementalStartingScanner(long start, long end) {
 
     @Override
     public Result scan(SnapshotManager manager, SnapshotReader reader) {
+        start = (start < manager.earliestSnapshotId()) ? manager.earliestSnapshotId() - 1 : start;
```

Comment:
reuse earliestSnapshotId and latestSnapshotId

Reply:
enen,I will modify to avoid reuse manager.earliestSnapshotId() and  manager.latestSnapshotId().And  any thing elase?